pkg/handlers/aws/mutation/securitygroups/inject.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package securitygroups
+
+import (
+	"context"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/d2iq-labs/capi-runtime-extensions/api/v1alpha1"
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/patches"
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/variables"
+	capav1 "github.com/d2iq-labs/capi-runtime-extensions/common/pkg/external/sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+)
+
+const (
+	// VariableName is the external patch variable name.
+	VariableName = "additionalSecurityGroups"
+)
+
+type awsSecurityGroupSpecPatchHandler struct {
+	metaVariableName  string
+	variableFieldPath []string
+	patchSelector     clusterv1.PatchSelector
+}
+
+func newAWSSecurityGroupSpecPatchHandler(
+	metaVariableName string,
+	variableFieldPath []string,
+	patchSelector clusterv1.PatchSelector,
+) *awsSecurityGroupSpecPatchHandler {
+	return &awsSecurityGroupSpecPatchHandler{
+		metaVariableName:  metaVariableName,
+		variableFieldPath: variableFieldPath,
+		patchSelector:     patchSelector,
+	}
+}
+
+func (h *awsSecurityGroupSpecPatchHandler) Mutate(
+	ctx context.Context,
+	obj *unstructured.Unstructured,
+	vars map[string]apiextensionsv1.JSON,
+	holderRef runtimehooksv1.HolderReference,
+	_ client.ObjectKey,
+) error {
+	log := ctrl.LoggerFrom(ctx).WithValues(
+		"holderRef", holderRef,
+	)
+	additionalSecGroupVar, found, err := variables.Get[v1alpha1.AdditionalSecurityGroup](
+		vars,
+		h.metaVariableName,
+		h.variableFieldPath...,
+	)
+	if err != nil {
+		return err
+	}
+	if !found {
+		log.V(5).
```

Comment:
nit
```suggestion
		log.V(5).Info("No additional security groups provided. Skipping.")
```

Reply:
This seems to have come from the linter so i will leave.